### DISCUSSION TITLE
Has anyone found a way to automate code generation for the Figma UI kit?

### DISCUSSION THREAD

**Discussion by bobg2023** [primefaces/community#967] [Figma UI Kit]
Hi, I’ve been playing around with Figma dev mode and some of the AI plugins like builder.io to see how far code generation has come. 
From my experience so far it’s not tge best in that it’s not ‘aware’ of prime react components that are represented by the Figma UI components. So it pretty much makes every a
DIV. 

I was wondering if anyone else has looked into this and has an approach to generate react code ( just the basic components and layout ) from Figma?

Thank you.

**Comment by onursenture** [comment]
Hello,

We are developing a product to streamline the theme creation process using a Figma UI Kit. With this product, users can quickly generate unique themes directly from Figma.

The current Sass-based styling mode will be replaced, and in its place, we're introducing a new theming API. This API will provide deep integration with Figma and facilitate direct theme generation from the design tool.

We also have plans to incorporate a Builder-like features into this product in the future.

**Reply by janeznavteh100** [reply]
Any ETA for these functions?

**Reply by onursenture** [reply]
We cannot provide an ETA at this moment.

**Reply by taylageben** [reply]
@onursenture is there any update on the ETA?

**Reply by onursenture** [reply]
> @onursenture is there any update on the ETA?

Theme Designer is out now for PrimeVue. Later PrimeNG and PrimeReact will follow up. https://primevue.org/designer/
